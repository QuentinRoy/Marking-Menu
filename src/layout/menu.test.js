import createMenu from './menu';

const createModel = (itemNb = 0) => ({
  children: Array.from({ length: itemNb }).map((_, i) => ({
    name: `item-${i}-name`,
    angle: i * 10,
    id: `item-${i}-id`,
  })),
});

describe('createMenu', () => {
  it('renders', () => {
    const div = document.createElement('div');
    createMenu(div, createModel(6), [30, 50], 'item-5-id', document);
    expect(div).toMatchSnapshot();
  });

  it('renders without active element', () => {
    const div = document.createElement('div');
    createMenu(div, createModel(4), [30, 50], undefined, document);
    expect(div).toMatchSnapshot();
  });

  it('update the active element', () => {
    const div = document.createElement('div');
    const m = createMenu(div, createModel(4), [30, 50], 'item-2-id', document);
    expect(div).toMatchSnapshot();
    m.setActive('item-1-id');
    expect(div).toMatchSnapshot();
  });

  it('can be removed', () => {
    const div = document.createElement('div');
    const m = createMenu(div, createModel(4), [30, 50], 'item-2-id', document);
    expect(div).toMatchSnapshot();
    m.remove();
    expect(div).toMatchSnapshot();
  });
});
