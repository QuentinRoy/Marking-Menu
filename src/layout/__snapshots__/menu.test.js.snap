// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMenu can be removed 1`] = `
<div>
  <div
    class="marking-menu"
    style="--center-x: 30px; --center-y: 50px;"
  >
    <div
      class="marking-menu-item"
      data-item-id="item-0-id"
      style="--angle: 0deg; --cosine: 1; --sine: 0;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-0-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-1-id"
      style="--angle: 10deg; --cosine: 0.984807753012208; --sine: -0.17364817766693033;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-1-name
      </div>
    </div>
    <div
      class="marking-menu-item active"
      data-item-id="item-2-id"
      style="--angle: 20deg; --cosine: 0.9396926207859084; --sine: -0.3420201433256687;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-2-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-3-id"
      style="--angle: 30deg; --cosine: 0.8660254037844387; --sine: -0.49999999999999994;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-3-name
      </div>
    </div>
  </div>
</div>
`;

exports[`createMenu can be removed 2`] = `<div />`;

exports[`createMenu renders 1`] = `
<div>
  <div
    class="marking-menu"
    style="--center-x: 30px; --center-y: 50px;"
  >
    <div
      class="marking-menu-item"
      data-item-id="item-0-id"
      style="--angle: 0deg; --cosine: 1; --sine: 0;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-0-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-1-id"
      style="--angle: 10deg; --cosine: 0.984807753012208; --sine: -0.17364817766693033;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-1-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-2-id"
      style="--angle: 20deg; --cosine: 0.9396926207859084; --sine: -0.3420201433256687;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-2-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-3-id"
      style="--angle: 30deg; --cosine: 0.8660254037844387; --sine: -0.49999999999999994;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-3-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-4-id"
      style="--angle: 40deg; --cosine: 0.766044443118978; --sine: -0.6427876096865393;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-4-name
      </div>
    </div>
    <div
      class="marking-menu-item active"
      data-item-id="item-5-id"
      style="--angle: 50deg; --cosine: 0.6427876096865394; --sine: -0.766044443118978;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-5-name
      </div>
    </div>
  </div>
</div>
`;

exports[`createMenu renders without active element 1`] = `
<div>
  <div
    class="marking-menu"
    style="--center-x: 30px; --center-y: 50px;"
  >
    <div
      class="marking-menu-item"
      data-item-id="item-0-id"
      style="--angle: 0deg; --cosine: 1; --sine: 0;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-0-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-1-id"
      style="--angle: 10deg; --cosine: 0.984807753012208; --sine: -0.17364817766693033;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-1-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-2-id"
      style="--angle: 20deg; --cosine: 0.9396926207859084; --sine: -0.3420201433256687;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-2-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-3-id"
      style="--angle: 30deg; --cosine: 0.8660254037844387; --sine: -0.49999999999999994;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-3-name
      </div>
    </div>
  </div>
</div>
`;

exports[`createMenu update the active element 1`] = `
<div>
  <div
    class="marking-menu"
    style="--center-x: 30px; --center-y: 50px;"
  >
    <div
      class="marking-menu-item"
      data-item-id="item-0-id"
      style="--angle: 0deg; --cosine: 1; --sine: 0;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-0-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-1-id"
      style="--angle: 10deg; --cosine: 0.984807753012208; --sine: -0.17364817766693033;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-1-name
      </div>
    </div>
    <div
      class="marking-menu-item active"
      data-item-id="item-2-id"
      style="--angle: 20deg; --cosine: 0.9396926207859084; --sine: -0.3420201433256687;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-2-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-3-id"
      style="--angle: 30deg; --cosine: 0.8660254037844387; --sine: -0.49999999999999994;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-3-name
      </div>
    </div>
  </div>
</div>
`;

exports[`createMenu update the active element 2`] = `
<div>
  <div
    class="marking-menu"
    style="--center-x: 30px; --center-y: 50px;"
  >
    <div
      class="marking-menu-item"
      data-item-id="item-0-id"
      style="--angle: 0deg; --cosine: 1; --sine: 0;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-0-name
      </div>
    </div>
    <div
      class="marking-menu-item active"
      data-item-id="item-1-id"
      style="--angle: 10deg; --cosine: 0.984807753012208; --sine: -0.17364817766693033;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-1-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-2-id"
      style="--angle: 20deg; --cosine: 0.9396926207859084; --sine: -0.3420201433256687;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-2-name
      </div>
    </div>
    <div
      class="marking-menu-item"
      data-item-id="item-3-id"
      style="--angle: 30deg; --cosine: 0.8660254037844387; --sine: -0.49999999999999994;"
    >
      <div
        class="marking-menu-line"
      />
      <div
        class="marking-menu-label"
      >
        item-3-name
      </div>
    </div>
  </div>
</div>
`;
